vault/acl.go

```diff
@@ -18,6 +18,14 @@ type ACL struct {
 	root bool
 }
 
+type aclEntry struct {
```

Comment:
This seems over complex. I would instead lean on type switching, where in the fast-path the radix tree holds the uint32 bitmap, and in the non-hotpath stores the `PathCapabilities` object directly.